code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -245,46 +272,321 @@
-	var/mob/living/carbon/xenomorph/X = owner
-	var/mob/living/carbon/victim = A
+	var/mob/living/victim = A
+	var/mob/living/carbon/xenomorph/impairer = owner //Type cast this for on_fire
 
 	if(!istype(victim))
+		to_chat(impairer, "<span class='xenodanger'>We can't impair this target!</span>")
 		return
 
 	if(victim.stat == DEAD)
+		to_chat(impairer, "<span class='xenodanger'>We can't impair the dead!</span>")
 		return
 
-	if(!can_use_ability(A, TRUE, override_flags = XACT_IGNORE_SELECTED_ABILITY))
+	if(impairer.on_fire)
+		to_chat(impairer, "<span class='xenodanger'>We're too busy being on fire to impair them!</span>")
+		return FALSE
+
+	var/distance = get_dist(impairer, victim)
+	if(distance > 5)
+		to_chat(impairer, "<span class='xenodanger'>They are too far for us to reach their minds! We must be [distance - 5] tiles closer!</spam>")
+		return FALSE
+
+	if(!impairer.line_of_sight(victim)) //Need line of sight.
+		to_chat(impairer, "<span class='xenowarning'>We require line of sight to impair them!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/impair_senses/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/victim = A
+
+	to_chat(X, "<span class='xenodanger'>We reach out into mind of [victim.name], impairing their senses...</span>") //Notify privately
+	to_chat(victim, "<span class='danger'>Your mind convulses at the touch of something ominous as the world seems to dim, going completely silent!</span>") //Notify privately
+	X.playsound_local(X, 'sound/effects/ghost.ogg', 25, 0, 1) //Spooky psychic noises.
+	victim.playsound_local(victim, 'sound/effects/ghost.ogg', 25, 0, 1) //Spooky psychic noises.
+	victim.blind_eyes(HUNTER_IMPAIR_SENSES_STACKS * 0.25) //Blind for a very short duration
+	victim.adjust_stagger(HUNTER_IMPAIR_SENSES_STACKS) //Stagger for a short duration
+	victim.blur_eyes(HUNTER_IMPAIR_SENSES_STACKS) //Blur for a short duration
+	victim.adjust_ear_damage(deaf = HUNTER_IMPAIR_SENSES_STACKS) //Temp deafen for a short duration
+
+	succeed_activate()
+	add_cooldown()
+
+	GLOB.round_statistics.hunter_impede_senses++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_impede_senses") //Statistics
+
+// ***************************************
+// *********** Sneak Stinger
+// ***************************************
+/datum/action/xeno_action/activable/sneak_stinger
+	name = "Stealth Stinger"
+	action_icon_state = "neuro_sting"
+	mechanics_text = "Use on an adjacent target while stealthed and your sneak attack is ready, while stationary or off run intent. Injects Acid while on Harm intent and Neurotoxin while on any other intent."
+	plasma_cost = 50
+	keybind_signal = COMSIG_XENOABILITY_STEALTH_STINGER
+	cooldown_timer = 30 SECONDS
+
+/datum/action/xeno_action/activable/sneak_stinger/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
```

Comment:
This if is unnecessary, it will runtime and stack trace on its own.

Reply:
Yes this